### ISSUE TITLE
Keep eol control char in multiline mode.

### ISSUE THREAD

**Issue by QuarkDoe** [aeshell/aesh-readline#44]
GNU readline keeps eol char when we read quoted string and remove it when ending line with backslash.
ex.
```
$ echo "test
> test"
test
test
$ echo test\
> test
testtest
```

But when i trying to use aesh readline it remove eol-char from each line in both cases.
ex.
```
$ "test
> test"
"testtest"
$ test\
> test
testtest
```

Test code:
```java
import org.aesh.readline.*;
import org.aesh.readline.editing.*;
import org.aesh.readline.terminal.DeviceBuilder;
import org.aesh.readline.tty.terminal.TerminalConnection;
import org.aesh.terminal.Connection;

import java.util.function.Consumer;

public class ReadLineTest implements Consumer<Connection>{

    public static void main( String[] args ){
        try{
            new TerminalConnection( new ReadLineTest() );
        }catch( Exception e ){
            e.printStackTrace();
        }
    }

    @Override
    public void accept( Connection connection ){
        EditModeBuilder builder = EditModeBuilder.builder( EditMode.Mode.EMACS );
        builder.device( DeviceBuilder.builder().name( System.getenv( "TERM" ) ).build() );
        read( connection, new Readline( builder.create() ) );
        connection.openBlocking();
    }

    private void read( Connection connection, Readline readline ){
        readline.readline( connection, new Prompt( "$ " ), input -> {
            if( input == null || "exit".equals( input ) ){
                connection.write( "we're exiting\n" );
                connection.close();
            }else{
                connection.write( input + "\n" );
                read( connection, readline );
            }
        });
    }

}

```

**Comment by stalep** [comment]
ah, thanks for the bug report, we'll look into it asap!

**Comment by stalep** [comment]
should be fixed now, we'll release 1.12 later today
